substrate/primitives/consensus/beefy/src/lib.rs

```diff
@@ -301,6 +301,37 @@ impl<Number, Id, Signature> EquivocationProof<Number, Id, Signature> {
 	}
 }
 
+/// Proof of authority misbehavior on a given set id.
+/// This proof shows commitment signed on a different fork.
+#[derive(Clone, Debug, Decode, Encode, PartialEq, TypeInfo)]
+pub struct ForkEquivocationProof<Number, Id, Signature, Header> {
```

Comment:
1. Nit: I would name this `CommitmentEquivocationProof`. Seems more consistent with `VoteEquivocationProof`.
2. Is there any advantage in being able to report commitment equivocations ? Couldn't we report each vote individually ?

Reply:
1. I like `CommitmentEquivocationProof` more than `ForkEquivocationProof` too, +1
2. yes, advantage is batching everything together - you get the conflicting proofs from the two sides of the bridge and build single `CommitmentEquivocationProof` from these two and submit in single extrinsic

Reply:
On 1:
I'm not yet convinced by `CommitmentEquivocationProof`: both votes and signed commitments contain commitments, so this introduces an ambiguity.
↦ Would you prefer `SignedCommitmentEquivocationProof` to `ForkEquivocationProof`? It's more verbose to avoid the above clash. 

The current naming refers to the origin of an equivocation (voting vs. forking) to distinguish between them. `ForkEquivocationProof` is also deliberately looser in naming since they can be detected also come from a `BeefyVersionedFinalityProof`, but that's a minor point since at this stage, `BeefyVersionedFinalityProof` is just a `SignedCommitment` wrapper.

On 2: 
Adrian already explained it well.

Reply:
1. I see your point. `SignedCommitmentEquivocationProof` seems too verbose. Thinking. Will get back if I have any other idea.

2. Thanks for the explanation ! Makes sense.

Reply:
IMO `SignedCommitmentEquivocationProof` is best option so far